repo/datastore.go

```diff
@@ -175,3 +177,32 @@ type Sales interface {
 	// Return the IDs for all orders
 	GetAll() ([]string, error)
 }
+
+type Cases interface {
+	// Save a new case
+	Put(caseID string, state pb.OrderState, buyerOpened bool, claim string) error
+
+	// Update a case with the buyer info
+	UpdateBuyerInfo(caseID string, buyerContract *pb.RicardianContract, buyerValidationErrors []string, buyerPayoutAddress string, buyerOutpoints []*pb.Outpoint) error
+
+	// Update a case with the vendor info
+	UpdateVendorInfo(caseID string, vendorContract *pb.RicardianContract, vendorValidationErrors []string, vendorPayoutAddress string, vendorOutpoints []*pb.Outpoint) error
+
+	// Mark a case as read in the database
+	MarkAsRead(caseID string) error
+
+	// Mark a case as closed in the database
+	MarkAsClosed(caseID string, resolution *pb.DisputeResolution) error
+
+	// Delete a case
+	Delete(caseID string) error
+
+	// Return the case metadata given a case ID
+	GetCaseMetadata(caseID string) (buyerContract, vendorContract *pb.RicardianContract, buyerValidationErrors, vendorValidationErrors []string, state pb.OrderState, read bool, timestamp time.Time, buyerOpened bool, claim string, resolution *pb.DisputeResolution, err error)
+
+	// Return the dispute payout data for a case
+	GetPayoutDetails(caseID string) (buyerContract, vendorContract *pb.RicardianContract, buyerPayoutAddress, vendorPayoutAddress string, buyerOutpoints, vendorOutpoints []*pb.Outpoint, state pb.OrderState, err error)
```

Comment:
I'll look into narrowing down the arguments after merge. I'll open an issue for it.